src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,205 @@
+                     + "WHERE id = ? AND is_deleted = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement
+                        = connection.prepareStatement(query)) {
+            statement.setString(1, car.getModel());
+            statement.setLong(2, car.getManufacturer().getId());
+            statement.setLong(3, car.getId());
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update " + car + " in DB", e);
+        }
+        deleteCarDrivers(car.getId());
+        setDrivers(car);
+        return car;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE cars SET is_deleted = TRUE WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement
+                        = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            return statement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete car by id " + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        List<Car> cars = new ArrayList<>();
+        String query = "SELECT c.id, c.model, m.id AS manufacturer_id, m.name, m.country "
+                     + "FROM cars c "
+                     + "JOIN cars_drivers cd ON c.id = cd.car_id "
+                     + "JOIN manufacturers m ON m.id = c.manufacturer_id "
+                     + "WHERE cd.driver_id = ? AND c.is_deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, driverId);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(parseCarWithManufacturerFromResultSet(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get cars by driver id " + driverId, e);
+        }
+        cars.forEach(car -> car.setDrivers(getDriversForCar(car.getId())));
+        return cars;
+    }
+
+    private Car parseCarWithManufacturerFromResultSet(ResultSet resultSet) throws SQLException {
+        Long carId = resultSet.getObject("id", Long.class);
+        String carModel = resultSet.getString("model");
+        Long manufacturerId = resultSet.getObject("manufacturer_id", Long.class);
+        String manufacturerName = resultSet.getString("name");
+        String manufacturerCountry = resultSet.getString("country");
+        Manufacturer carManufacturer = new Manufacturer(manufacturerName, manufacturerCountry);
+        carManufacturer.setId(manufacturerId);
+        Car car = new Car(carModel, carManufacturer, getDriversForCar(carId));
```

Comment:
```suggestion
        Car car = new Car(carModel, carManufacturer);
```
тут не варто викликати метод який дістає драйверів. Адже виходить що в нас буде відкриватись конекшин в конекшині а це не добре при роботі з БД